src/components/KCodeBlock/KCodeBlockIconButton.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <KTooltip
+    class="code-block-action-tooltip"
+    placement="bottomStart"
+    :popover-timeout="300"
+    :text="copyTooltipText"
+  >
+    <KButton
+      appearance="tertiary"
+      class="code-block-action-button"
+      :class="[`theme-${theme}`, { 'code-block-action-button-active': active }]"
+      v-bind="attrs"
+      @click="onClick"
+    >
+      <!-- TODO: [beta] update once icon slot is deprecated -->
+      <template #icon>
+        <slot />
+      </template>
```

Comment:
**Question**: Shouldn’t this just re-expose the button’s default slot? 

```suggestion
      <slot />
```

Seems simpler and I don’t think anyone would need it to be the icon slot specifically currently.

Reply:
We want it to be the icon slot so that the button is square (padding top and bottom = padding left and right). Otherwise it will look like this.

![Screenshot 2024-04-04 at 9 03 37 AM](https://github.com/Kong/kongponents/assets/36751160/f34c62a7-23e3-4413-a1b8-f702fd8f7abf)

Reply:
Isn't that going to make us have to do another followup PR to change it in a few weeks..?

Reply:
Yes but I'm planning to make it backwards compatible